src/simix/smx_network.cpp

```diff
@@ -582,6 +594,8 @@ void SIMIX_comm_finish(smx_synchro_t synchro)
       continue; // if process handling comm is killed
     if (simcall->call == SIMCALL_COMM_WAITANY) {
       SIMIX_waitany_remove_simcall_from_actions(simcall);
+      if (simcall->timer) 
+        SIMIX_timer_remove(simcall->timer);
```

Comment:
I think we should set `simcall->timer` to `nullptr` here in order to avoid letting some dangling pointer.